Xwt.XamMac/Xwt.Mac/WindowFrameBackend.cs

```diff
@@ -334,20 +331,14 @@ protected virtual void OnBoundsChanged ()
 
 		void IWindowFrameBackend.SetTransientFor (IWindowFrameBackend parent)
 		{
+			//TODO: why this?
 			if (!((IWindowFrameBackend)this).ShowInTaskbar)
 				Window.StyleMask &= ~NSWindowStyle.Miniaturizable;
 
-			var win = Window as NSWindow ?? ApplicationContext.Toolkit.GetNativeWindow (parent) as NSWindow;
-
-			if (Window.ParentWindow != win) {
-				// remove from the previous parent
-				if (Window.ParentWindow != null)
-					Window.ParentWindow.RemoveChildWindow (Window);
```

Comment:
I do not know this code very well. One thing we are not doing now is the above code ^ Is that a problem?

Trying to think of a scenario - maybe make the window visible, hide it, then change the parent and make it visible again? Would you have to remove it from the old parent before adding it to the new one?

Reply:
oh true! good catch Matt!

Reply:
fixed